osquery/core/windows/bstr.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/utils/system/system.h>
+
+#include <string>
+
+#include <wtypes.h>
+
+#include <osquery/utils/only_movable.h>
+
+namespace osquery {
+
+class Bstr : private only_movable {
+ public:
+  static Bstr fromString(std::wstring_view);
+
+  Bstr() = default;
+  ~Bstr();
```

Comment:
Destructor and other methods that don't throw exceptions could be marked with `noexcept` keyword

Reply:
> Also IIRC, the destructor is always has implicit `noexcept` spec by default

Thanks for sharing this! I was not aware of it. I agree that adding `noexcept` to other methods would be beneficial.